common/app/common/Logback/KinesisAdapter.scala

```diff
@@ -0,0 +1,49 @@
+                        package common.Logback
+
+import java.util.concurrent.ThreadPoolExecutor
+
+import akka.actor.ActorSystem
+import akka.dispatch.MessageDispatcher
+import akka.pattern.CircuitBreaker
+import ch.qos.logback.classic.spi.ILoggingEvent
+import com.amazonaws.ClientConfiguration
+import com.amazonaws.auth.AWSCredentialsProvider
+import com.amazonaws.retry.{PredefinedRetryPolicies, RetryPolicy}
+import com.amazonaws.services.kinesis.AmazonKinesisAsyncClient
+import com.gu.logback.appender.kinesis.KinesisAppender
+import scala.concurrent.duration._
+import scala.concurrent.Future
+
+// LogbackOperationsPool must be wired as a singleton
+class LogbackOperationsPool(val actorSystem: ActorSystem)  {
+  val logbackOperations: MessageDispatcher = actorSystem.dispatchers.lookup("akka.logback-operations")
+}
+
+class SafeBlockingKinesisAppender(logbackOperations: LogbackOperationsPool) extends KinesisAppender[ILoggingEvent] {
+
+  private val breaker = new CircuitBreaker(
+    logbackOperations.actorSystem.scheduler,
+    maxFailures = 1,
+    callTimeout = 1.seconds,
+    resetTimeout = 10.seconds
+  )(logbackOperations.logbackOperations)
+
+  override protected def createClient(credentials: AWSCredentialsProvider, configuration: ClientConfiguration, executor: ThreadPoolExecutor): AmazonKinesisAsyncClient = {
+    configuration.setMaxErrorRetry(0)
+    configuration.setRetryPolicy(
+      new RetryPolicy(PredefinedRetryPolicies.NO_RETRY_POLICY.getRetryCondition, PredefinedRetryPolicies.DEFAULT_BACKOFF_STRATEGY, 0, true)
+    )
+    new AmazonKinesisAsyncClient(credentials, configuration, executor)
+  }
+
+  override protected def putMessage(message: String): Unit = {
+    breaker.withCircuitBreaker {
+      Future {
+        super.putMessage(message)
+      }(logbackOperations.logbackOperations)
```

Comment:
It's probably worth commenting here, as some keen person might remove or refactor out the non-implicit EC in future otherwise. At the least, it will help people understand it's importance.